app/models/async_transaction/base.rb

```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+module AsyncTransaction
+  class Base < ActiveRecord::Base
+    self.table_name = 'async_transactions'
+    validates :id, uniqueness: true
+    validates :user_uuid, :source_id, :source, :status, :transaction_id, presence: true
+    validates :transaction_id, uniqueness: { scope: :source }
```

Comment:
Can we add a custom message option in here, that would explain to the end user precisely what validation is being broken.